drivers/gpio/gpio_stm32.c

```diff
@@ -37,6 +36,226 @@ static void gpio_stm32_isr(int line, void *arg)
+		} else {
+			LL_GPIO_SetPinMode(gpio, pin, LL_GPIO_MODE_INPUT);
+
+			temp = conf & (STM32_PUPD_MASK << STM32_PUPD_SHIFT);
+
+			if (temp == STM32_PUPD_PULL_UP) {
+				LL_GPIO_SetPinPull(gpio, pin, LL_GPIO_PULL_UP);
+			} else {
+				LL_GPIO_SetPinPull(gpio, pin, LL_GPIO_PULL_DOWN);
+			}
+		}
+
+	} else {
+		temp = conf & (STM32_CNF_OUT_1_MASK << STM32_CNF_OUT_1_SHIFT);
+
+		if (temp == STM32_CNF_GP_OUTPUT) {
+			LL_GPIO_SetPinMode(gpio, pin, LL_GPIO_MODE_OUTPUT);
+		} else {
+			LL_GPIO_SetPinMode(gpio, pin, LL_GPIO_MODE_ALTERNATE);
+		}
+
+		temp = conf & (STM32_CNF_OUT_0_MASK << STM32_CNF_OUT_0_SHIFT);
+
+		if (temp == STM32_CNF_PUSH_PULL) {
+			LL_GPIO_SetPinOutputType(gpio, pin, LL_GPIO_OUTPUT_PUSHPULL);
+		} else {
+			LL_GPIO_SetPinOutputType(gpio, pin, LL_GPIO_OUTPUT_OPENDRAIN);
+		}
+
+		temp = conf & (STM32_MODE_OSPEED_MASK << STM32_MODE_OSPEED_SHIFT);
+
+		if (temp == STM32_MODE_OUTPUT_MAX_2) {
+			LL_GPIO_SetPinSpeed(gpio, pin, LL_GPIO_SPEED_FREQ_LOW);
+		} else if (temp == STM32_MODE_OUTPUT_MAX_10) {
+			LL_GPIO_SetPinSpeed(gpio, pin, LL_GPIO_SPEED_FREQ_MEDIUM);
+		} else {
+			LL_GPIO_SetPinSpeed(gpio, pin, LL_GPIO_SPEED_FREQ_HIGH);
+		}
+	}
+#else
+	u32_t temp = conf & (STM32_MODER_MASK << STM32_MODER_SHIFT);
+
+	if (temp == STM32_MODER_OUTPUT_MODE) {
+		LL_GPIO_SetPinMode(gpio, pin, LL_GPIO_MODE_OUTPUT);
+
+		temp = conf & (STM32_OTYPER_MASK << STM32_OTYPER_SHIFT);
+
+		if (temp == STM32_OTYPER_PUSH_PULL) {
+			LL_GPIO_SetPinOutputType(gpio, pin, LL_GPIO_OUTPUT_PUSHPULL);
+		} else {
+			LL_GPIO_SetPinOutputType(gpio, pin, LL_GPIO_OUTPUT_OPENDRAIN);
+		}
+
+		temp = conf & (STM32_OSPEEDR_MASK << STM32_OSPEEDR_SHIFT);
+
+		if (temp == STM32_OSPEEDR_LOW_SPEED) {
+			LL_GPIO_SetPinSpeed(gpio, pin, LL_GPIO_SPEED_FREQ_LOW);
+		} else if (temp == STM32_OSPEEDR_MEDIUM_SPEED) {
+			LL_GPIO_SetPinSpeed(gpio, pin, LL_GPIO_SPEED_FREQ_MEDIUM);
+		} else if (temp == STM32_OSPEEDR_HIGH_SPEED) {
+			LL_GPIO_SetPinSpeed(gpio, pin, LL_GPIO_SPEED_FREQ_HIGH);
+#if defined(CONFIG_SOC_SERIES_STM32F2X) || defined(CONFIG_SOC_SERIES_STM32F4X) || \
+	defined(CONFIG_SOC_SERIES_STM32F7X) || defined(CONFIG_SOC_SERIES_STM32L0X) || \
+	defined(CONFIG_SOC_SERIES_STM32L4X)
+		} else if (temp == STM32_OSPEEDR_VERY_HIGH_SPEED) {
+			LL_GPIO_SetPinSpeed(gpio, pin, LL_GPIO_SPEED_FREQ_VERY_HIGH);
+#endif
+		} else {
+			return -EINVAL;
+		}
+	} else if (temp == STM32_MODER_INPUT_MODE) {
+		LL_GPIO_SetPinMode(gpio, pin, LL_GPIO_MODE_INPUT);
+
+		temp = conf & (STM32_PUPDR_MASK << STM32_PUPDR_SHIFT);
+
+		if (temp == STM32_PUPDR_PULL_UP) {
+			LL_GPIO_SetPinPull(gpio, pin, LL_GPIO_PULL_UP);
+		} else if (temp == STM32_PUPDR_PULL_DOWN) {
+			LL_GPIO_SetPinPull(gpio, pin, LL_GPIO_PULL_DOWN);
+		} else {
+			LL_GPIO_SetPinPull(gpio, pin, LL_GPIO_PULL_NO);
+		}
+	} else if (temp == STM32_MODER_ANALOG_MODE) {
+		LL_GPIO_SetPinMode(gpio, pin, LL_GPIO_MODE_ANALOG);
+	} else {
+		LL_GPIO_SetPinMode(gpio, pin, LL_GPIO_MODE_ALTERNATE);
+
+		if (pin < (1 << 8)) {
```

Comment:
This is correct but confusing. 
We should introduce an alternate variable for pin used with LL api  (pin_ll ? Anything better is welcome) so we could use a simple and clear if statement here (if pin < 8).